packages/md-enhance/src/client/components/playground/Playground.ts

```diff
@@ -0,0 +1,64 @@
+import { defineComponent, h } from "vue";
+import type { VNode } from "vue";
+
+import { ClientOnly } from "@vuepress/client";
+
+import PlaygroundExternal from "./PlaygroundExternal";
+import PlaygroundInternal from "./PlaygroundInternal";
+import PlayFile from "./PlayFile";
+import PlayImports from "./PlayImports";
+import PlaySettings from "./PlaySettings";
+
+import { parsePlaygroundSettings } from "../../utils/playground";
+import { IMPORT_MAP_KEY } from "../../../shared/playground";
+
+export default defineComponent({
+  // eslint-disable-next-line vue/multi-word-component-names
+  name: "Playground",
+
+  components: {
+    PlayFile,
```

Comment:
If you are using render function you do not need to declare components here, so you should remove it.

Also I suggested names like ExternalPlayground and PlaygroundFile